chain/network/src/routing.rs

```diff
@@ -739,8 +741,19 @@ impl RoutingTable {
         }
     }
 
-    pub fn get_raw_graph(&self) -> &HashMap<PeerId, HashSet<PeerId>> {
-        &self.raw_graph.adjacency
+    pub fn get_raw_graph(&self) -> HashMap<PeerId, HashSet<PeerId>> {
```

Comment:
looks like this function is only used when we enable `metric_recorder`. It might be better to guard the function with the feature so that we don't use it anywhere else (as it is expensive)